src/brokerbot/BrokerbotRouter.sol

```diff
@@ -46,8 +47,7 @@ contract BrokerbotRouter is ISwapRouter {
 	) external payable override checkDeadline(params.deadline) returns (uint256 amountIn) {
 		(IBrokerbot brokerbot, PaymentHub paymentHub) = BrokerbotLib.getBrokerbotAndPaymentHub(brokerbotRegistry, IERC20(params.tokenIn), IERC20(params.tokenOut));
 		if (msg.value > 0) {
-			amountIn = _buyWithEther(brokerbot, paymentHub, params.amountOut, msg.value);
-			refundETH();
+			revert ETH_Swap_Not_Supported();
```

Comment:
I would move this check now to the beginning of the function to safe a little gas and move the else part into the function directly.
Also this comment `@dev Single swap with ETH works, but base currency address has to be provided as tokenIn.` isn't correct anymore